backend/entityservice/database/selections.py

```diff
@@ -274,19 +275,42 @@ def get_bloomingdata_columns(db, dp_id, columns):
     return [result[column] for column in columns]
 
 
+def get_encodingblock_ids(db, dp_id, block_name=None):
+    """Yield all encoding ids in either a single block, or all blocks for a given data provider."""
+    sql_query = """
+        SELECT encoding_id 
+        FROM encodingblocks
+        WHERE dp = %s
+        {}
+        """.format("AND block_id = %s" if block_name else "")
+    # Specifying a name for the cursor creates a server-side cursor, which prevents all of the
+    # records from being downloaded at once.
+    cur = db.cursor(f'encodingfetcher-{dp_id}')
+
+    args = (dp_id, block_name) if block_name else (dp_id,)
+
+    cur.execute(sql_query, args)
+    while True:
+        rows = cur.fetchmany(10_000)
+        if not rows:
+            break
+        for row in rows:
+            yield row[0]
```

Comment:
Why don't you think it would be good to keep the db connection while streaming through the blocks? Establishing a db connection is not free. The point of this sneaky Python cache is because we might not have the memory to store all (e.g. millions) of blocks if we used `fetchall()`, but the network overhead of `fetchone` if we fetched and yielded each row one at a time would be a killer.